### PR TITLE
Fix for /devicesview endpoint not returing device group info.

### PR DESCRIPTION
**Pull request by AaronH88**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
8efcc384-6db5-4745-91b0-748ed1e13ed4
e448cfe2-ec3f-44ea-8f41-ac1feb406179

File: pkg/services/devices.go [hunk 1]
```diff
@@ -622,17 +622,18 @@ func (s *DeviceService) GetDevicesView(limit int, offset int, tx *gorm.DB) (*mod
...
 	for _, device := range storedDevices {
+		var tempArr []models.DeviceDeviceGroup
 		for _, deviceGroups := range device.DevicesGroups {
-			if _, ok := deviceToGroupMap[device.ID]; ok {
-				temp := models.DeviceDeviceGroup{ID: deviceGroups.ID, Name: deviceGroups.Name}
-				deviceToGroupMap[device.ID].info = append(deviceToGroupMap[device.ID].info, temp)
-			}
+			tempArr = append(tempArr, models.DeviceDeviceGroup{ID: deviceGroups.ID, Name: deviceGroups.Name})
 		}
+		deviceToGroupMap[device.ID] = &deviceGroupInfo{info: tempArr}
```
Comment (ldjebran) [inline pkg/services/devices.go:636]: maybe we do need this deviceGroupInfo and directlly set
```go 
deviceToGroupMap := make(map[uint][]models.DeviceDeviceGroup)

...
deviceToGroupMap[device.ID] =  tempArr

// line 691
               if _, ok := deviceToGroupMap[device.ID]; ok {
			deviceGroups = deviceToGroupMap[device.ID]
		}

```

what do you think ?

Comment (AaronH88) [reply pkg/services/devices.go:636]: Makes sense!

changing now, thanks!

**Review by ldjebran — APPROVED** [review]
LGMT

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
26cd4833-4c43-4e7e-aeeb-054a6810f836
cbc36d6c-a4ee-47f5-b23c-931d51c7f4ff

**Pull request merged**